src/types.rs

```diff
@@ -47,27 +48,63 @@ impl LintPass for TypePass {
 }
 
 impl LateLintPass for TypePass {
-    fn check_ty(&mut self, cx: &LateContext, ast_ty: &Ty) {
```

Comment:
I tried that but while `cx.tcx.map.get_parent(ast_ty.id)` does not return `DUMMY_NODE_ID`, `cx.tcx.map.find(cx.tcx.map.get_parent(ast_ty.id))` is always `None` here.